lib/chef/provider/mount/mount.rb

```diff
@@ -268,6 +247,35 @@ def mount_options_unchanged?
             @current_resource.pass == @new_resource.pass
         end
 
+        # It will update or delete the entry from fstab.
+        def edit_fstab(action)
```

Comment:
this should not be called `action` since that collides with the `action` on the resource, and `:update` and `:remove` are not a valid actions for the mount provider.  different concepts must have different names.

you can just call this `def edit_fstab(remove: false)` and then do `if remove` later, then call it with `edit_fstab(remove: true)` with a named parameter to make it clear what is happening.

Reply:
@lamont-granquist I update this, please have a look.